integration_tests/multi_components.cpp

```diff
@@ -0,0 +1,65 @@
+/**
+  @file Integration test: Handling Multiple components.
+  @author Shakthi Prashanth <shakthi.prashanth.m@intel.com>
+*/
+
+#include <iostream>
+#include "integration_test_helper.h"
+#include "dronecore.h"
+#include "system.h"
+
+using namespace dronecore;
+using namespace std::this_thread;
+using namespace std::chrono;
+
+/**
+ * @brief This test detects ALL the components found across systems and tells
+ * whether each system is a standalone or the one with a system with componets
+ * such as autopilot with camera(s), etc. One way to run this test is as below:
+ * /////////////////////////////////////
+ * 1. Connect a V4L2 Camera on Ubuntu.
+ * 2. Launch Camera Streaming Daemon on Ubuntu and configure UDP port 14550.
+ * 3. Launch PX4 SITL.
+ * 4. Run SitlTest.MultiComponentDiscovery test.
+ * /////////////////////////////////////
+ */
+TEST_F(SitlTest, MultiComponentDiscovery)
+{
+    DroneCore dc;
+
+    // For both Autopilot and Camera
+    ASSERT_EQ(dc.add_udp_connection(), ConnectionResult::SUCCESS);
+
+    // FIXME: As components send Heartbeats at 1Hz,
+    // lets wait until atleast 2 of them gets discovered.
+    // Without this wait, we MAY NOT get Hearbeat from second component.
+    std::cout << "We've nothing to do until we've some system to talk to. Lets sleep!\n";
+    sleep_for(seconds(3));
+
+    auto uuids = dc.system_uuids();
+
+    for (auto uuid : uuids) {
+        std::cout << "We found a System with UUID: " << uuid << '\n';
+        System &system = dc.system(uuid);
+
+        auto is_autopilot = system.is_autopilot();
+
+        auto is_standalone = system.is_standalone();
+
+        auto has_camera = system.has_camera(); // by default checks for camera 1
```

Comment:
I want to ask few suggestions. I'd done testing of this by attaching a V4L2 camera on Ubuntu.
But, when I tested it today with a new camera, its component ID turned out to be `101` (usually it was `100`).
The test queries **whether we discovered camera 1** ? This time we discovered camera 2. As the test didn't discover camera 1, it prints as *Its an Autopilot alone.* 

It should rather query **did we discover any camera ?** If yes, continue the rest of the test.

We need a method that returns `true` if the system has at least one camera. We have two options:
- Add a new method `has_any_camera()`.
- Overload `has_camera()`, when 0 is passed to know whether system has at least a camera.
What shall we do ?

Reply:
Interesting question, but I would not change it in this PR. What about opening an issue to discuss this point and make a new PR later with a fix for it? :-)

Reply:
Later is fine. I think `has_camera` without argument should be `true` if it has any camera.

Reply:
@JonasVautherin I agree.
@julianoes That would break our original idea. We thought that `has_camera()`, by default should check for camera 1 (to avoid passing 1 every time). I think, `has_camera(0)` should be OK because, by concept: Comp ID 0 indicates all IDs. So it means check for any camera on the system.

Reply:
@julianoes I think what you suggested makes sense.